doc: add a git section in tool and config section (#5774)
nrfconnect/sdk-nrf: doc/nrf/introduction.rst

* Ninja (comparable to make) uses the build files to build the program, see the `Ninja documentation`_.
* The `GCC compiler`_ creates the executables.

Git
===

`Git`_ is a free and open source distributed version control system that allows managing the changes in the code or other collections of information (set of files) over time.

Git organizes data (files or directories) in project repositories.
The data is managed like a series of snapshots.
Every time you commit, or save the state of your project, Git takes a snapshot of what the files look like at that exact moment and stores a reference to that snapshot.
For unchanged files, Git provides just a link to the previous identical file it has already stored.

Git offers a lot of flexibility in how users manage changes, and repositories are easily duplicated.

> In |NCS|, the forking workflow is the agreed-upon Git workflow.

Reviewer: Maybe just: ... forking is the agree-upon Git workflow.

To contribute, the official public repository in GitHub is forked.

> When you say you are *forking* a repository you are creating a copy of the repository under your GitHub ID.

Reviewer: ```suggestion
When you say you are *forking* a repository, you are creating a copy of the repository under your GitHub ID.
```

This means that you are creating an identical copy that might diverge from the original over time.
This copy is your personal public repository that nobody else is allowed to push to, but changes can be pulled from it.

The original repository is called the *upstream* repository, and the newly created copy the *downstream* repository.

> Any changes made to the original repository will be reflected back to your forked repositories by using fetch and rebase commands.

Reviewer: ```suggestion
Any changes made to the original repository are reflected back to your forked repositories by using fetch and rebase commands.
```
